lib/gitter-adapter.js

```diff
@@ -216,10 +219,25 @@ Adapter.prototype.subscribeToRoom = function(room) {
 };
 
 Adapter.prototype.sendMessage = function(target, message) {
+  this.sendQueue.push({"target": target, "message": message});
+  this.checkSendQueue();
+};
+
+Adapter.prototype.checkSendQueue = function() {
+  if (!this.sending && this.sendQueue.length >= 1) {
+    var message = this.sendQueue.shift();
+    this.actualSendMessage(message["target"], message["message"]);
```

Comment:
I think we should go with a promise chain here instead of making an odd recursive loop.

```js
this.sendQueue.reduce(function(promiseChain, entry) {
  return promiseChain.then(function() {
  	return this.sendMessage(message['target'], message['message']);
  });
}, Promise.resolve());
```